### ISSUE TITLE
AttributeError: module 'griffe.expressions' has no attribute 'Name'

### ISSUE THREAD

**Issue by tomicapretto** [machow/quartodoc#242]
Hi! First of all, thanks for this great package. I'm looking forward to migrate from sphinx to quarto and this looks excellent.

I got started with the example from the webpage. But I can't get `quartodoc build` to work. I'm using the development version. See that it tries to look for `Name` within `griffe.expressions` and it's not there. I also see there's a TODO comment in line 124 of `md_renderer.py`. 

```python
(bambi) tomas@tomas:~/Desktop/OSS/bambinos/bambi/qdocs$ quartodoc build
Traceback (most recent call last):
  File "/home/tomas/anaconda3/envs/bambi/bin/quartodoc", line 8, in <module>
    sys.exit(cli())
  File "/home/tomas/anaconda3/envs/bambi/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/home/tomas/anaconda3/envs/bambi/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/tomas/anaconda3/envs/bambi/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/tomas/anaconda3/envs/bambi/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/tomas/anaconda3/envs/bambi/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/home/tomas/anaconda3/envs/bambi/lib/python3.10/site-packages/quartodoc/__main__.py", line 177, in build
    doc_build()
  File "/home/tomas/anaconda3/envs/bambi/lib/python3.10/site-packages/quartodoc/autosummary.py", line 506, in build
    self.write_doc_pages(pages, filter)
  File "/home/tomas/anaconda3/envs/bambi/lib/python3.10/site-packages/quartodoc/autosummary.py", line 540, in write_doc_pages
    rendered = self.renderer.render(page)
  File "/home/tomas/anaconda3/envs/bambi/lib/python3.10/site-packages/plum/function.py", line 483, in __call__
    return self._f(self._instance, *args, **kw_args)
  File "/home/tomas/anaconda3/envs/bambi/lib/python3.10/site-packages/plum/function.py", line 393, in __call__
    return _convert(method(*args, **kw_args), return_type)
  File "/home/tomas/anaconda3/envs/bambi/lib/python3.10/site-packages/quartodoc/renderers/md_renderer.py", line 186, in render
    return "\n\n".join([*header, *result])
  File "/home/tomas/anaconda3/envs/bambi/lib/python3.10/site-packages/plum/function.py", line 483, in __call__
    return self._f(self._instance, *args, **kw_args)
  File "/home/tomas/anaconda3/envs/bambi/lib/python3.10/site-packages/plum/function.py", line 393, in __call__
    return _convert(method(*args, **kw_args), return_type)
  File "/home/tomas/anaconda3/envs/bambi/lib/python3.10/site-packages/quartodoc/renderers/md_renderer.py", line 305, in render
    meth_docs.extend([self.render(x) for x in raw_meths if isinstance(x, layout.Doc)])
  File "/home/tomas/anaconda3/envs/bambi/lib/python3.10/site-packages/quartodoc/renderers/md_renderer.py", line 305, in <listcomp>
    meth_docs.extend([self.render(x) for x in raw_meths if isinstance(x, layout.Doc)])
  File "/home/tomas/anaconda3/envs/bambi/lib/python3.10/site-packages/plum/function.py", line 483, in __call__
    return self._f(self._instance, *args, **kw_args)
  File "/home/tomas/anaconda3/envs/bambi/lib/python3.10/site-packages/plum/function.py", line 393, in __call__
    return _convert(method(*args, **kw_args), return_type)
  File "/home/tomas/anaconda3/envs/bambi/lib/python3.10/site-packages/quartodoc/renderers/md_renderer.py", line 314, in render
    return "\n\n".join([title, self.render(el.obj)])
  File "/home/tomas/anaconda3/envs/bambi/lib/python3.10/site-packages/plum/function.py", line 483, in __call__
    return self._f(self._instance, *args, **kw_args)
  File "/home/tomas/anaconda3/envs/bambi/lib/python3.10/site-packages/plum/function.py", line 393, in __call__
    return _convert(method(*args, **kw_args), return_type)
  File "/home/tomas/anaconda3/envs/bambi/lib/python3.10/site-packages/quartodoc/renderers/md_renderer.py", line 336, in render
    body = self.render(section)
  File "/home/tomas/anaconda3/envs/bambi/lib/python3.10/site-packages/plum/function.py", line 483, in __call__
    return self._f(self._instance, *args, **kw_args)
  File "/home/tomas/anaconda3/envs/bambi/lib/python3.10/site-packages/plum/function.py", line 393, in __call__
    return _convert(method(*args, **kw_args), return_type)
  File "/home/tomas/anaconda3/envs/bambi/lib/python3.10/site-packages/quartodoc/renderers/md_renderer.py", line 431, in render
    rows = list(map(self.render, el.value))
  File "/home/tomas/anaconda3/envs/bambi/lib/python3.10/site-packages/plum/function.py", line 483, in __call__
    return self._f(self._instance, *args, **kw_args)
  File "/home/tomas/anaconda3/envs/bambi/lib/python3.10/site-packages/plum/function.py", line 393, in __call__
    return _convert(method(*args, **kw_args), return_type)
  File "/home/tomas/anaconda3/envs/bambi/lib/python3.10/site-packages/quartodoc/renderers/md_renderer.py", line 441, in render
    annotation = self.render_annotation(el.annotation)
  File "/home/tomas/anaconda3/envs/bambi/lib/python3.10/site-packages/quartodoc/renderers/md_renderer.py", line 125, in render_annotation
    if isinstance(el, expr.Name):
AttributeError: module 'griffe.expressions' has no attribute 'Name'
```

I added a `print()` to the `render_annotation` method and figured out it fails when `el` is an instance of `ExprName`

```python
    def render_annotation(self, el: "str | expr.Name | expr.Expression | None"):
        """Special hook for rendering a type annotation.

        Parameters
        ----------
        el:
            An object representing a type annotation.
            
        """
	
        print(type(el), el)
	
        if isinstance(el, (type(None), str)):
            return el

        # TODO: maybe there is a way to get tabulate to handle this?
        # unescaped pipes screw up table formatting
        if isinstance(el, expr.Name):
            return sanitize(el.source)
```

```
<class 'NoneType'> None
<class 'str'> None
<class 'NoneType'> None
<class 'NoneType'> None
<class 'NoneType'> None
<class 'NoneType'> None
<class 'NoneType'> None
<class 'NoneType'> None
<class 'NoneType'> None
<class 'NoneType'> None
<class 'NoneType'> None
<class 'NoneType'> None
<class 'NoneType'> None
<class 'NoneType'> None
<class 'NoneType'> None
<class 'NoneType'> None
<class 'griffe.expressions.ExprName'> int
Traceback (most recent call last):...
```

So then I changed the `if` statement to the following:

```python
        if isinstance(el, expr.ExprName):
            return sanitize(el.source)
```

and now I get `AttributeError: 'ExprName' object has no attribute 'source'`. However, when I go to the source code,

https://github.com/mkdocstrings/griffe/blob/6b943c8fc8e8dcfdc2a0fc9c11a210265b603be7/src/griffe/expressions.py#L440-L446

```python
@dataclass(eq=False, **dataclass_opts)
class ExprName(Expr):
    """This class represents a Python object identified by a name in a given scope.


    Attributes:
        source: The name as written in the source code.
    """
```

it is supposed to be there (according to the docstring)... So, my question is: are you aware of this? Do you think it's something that should be handle here in quartodoc, or should I post this on `griffe`? 

Thank you!

**Comment by tomicapretto** [comment]
Update:

I modified 

https://github.com/machow/quartodoc/blob/772feed768e0be74ed99391b92a4ef1c416b1527/quartodoc/renderers/md_renderer.py#L110C1-L128C33

to look like

```python
    def render_annotation(self, el: "str | expr.Name | expr.Expression | None"):
        """Special hook for rendering a type annotation.

        Parameters
        ----------
        el:
            An object representing a type annotation.
            
        """
	
        if isinstance(el, (type(None), str)):
            return el

        # TODO: maybe there is a way to get tabulate to handle this?
        # unescaped pipes screw up table formatting
        if isinstance(el, expr.ExprName):
            return sanitize(el.name)

        return sanitize(el.path)
```

and now it worked (replaced `expr.Name` with `expr.ExprName` and `el.source` with `el.path`)

**Comment by tomicapretto** [comment]
I think this is where things changed in `griffe`

https://github.com/mkdocstrings/griffe/commit/66c8ad5074e1475aa88a51d8652b5e197760d774

**Comment by machow** [comment]
Hey, thanks for raising! I'm just getting back from vacation, and will work on getting things working with the latest griffe release (the author was kind enough to flag early in [this discussion](https://github.com/mkdocstrings/griffe/discussions/195)).

In the meantime, if you pin to `griffe==0.32.3` (or lower) does it work? (I just checked the tests against `v0.32.3` and they passed; I'll create a release with an upper bound on its version or fix in the next day)

**Comment by tomicapretto** [comment]
Thanks for the prompt reply! I'm happy to open a PR if you want some help :) Otherwise, pinning dependencies will work. Thanks!

**Comment by tomicapretto** [comment]
@machow I think this is fixed, no? I'm working with `quartodoc==0.6.1` and it works perfectly. Feel free to close the issue. Thanks for the great library!

**Comment by machow** [comment]
Hey, glad it's working! I pinned the version of griffe for now, but will loop back and handle supporting newer versions after posit conf!
